### DISCUSSION TITLE
trying to make a character (first time)

### DISCUSSION THREAD

**Discussion by MryeetFN** [JordanSantiagoYT/FNF-JS-Engine#470] [Q&A] [answered]
so i keep getting this error over and over and i have no idea how to fix it.

C:\hostedtoolcache\windows\haxe\4.3.4\x64\std/haxe/format/JsonParser.hx (Line 318)
C:\hostedtoolcache\windows\haxe\4.3.4\x64\std/haxe/format/JsonParser.hx (Line 110)
C:\hostedtoolcache\windows\haxe\4.3.4\x64\std/haxe/format/JsonParser.hx (Line 90)
C:\hostedtoolcache\windows\haxe\4.3.4\x64\std/haxe/format/JsonParser.hx (Line 57)
Character.hx (Line 152)
editors/CharacterEditorState.hx (Line 1060)
editors/CharacterEditorState.hx (Line 377)
FlxUIDropDownMenuCustom.hx (Line 512)
FlxUIDropDownMenuCustom.hx (Line 361)
flixel/ui/FlxButton.hx (Line 521)
flixel/addons/ui/FlxUITypedButton.hx (Line 1182)
flixel/addons/ui/FlxUIButton.hx (Line 362)
flixel/ui/FlxButton.hx (Line 507)
openfl/events/EventDispatcher.hx (Line 426)
openfl/display/DisplayObject.hx (Line 1398)

Uncaught Error: Invalid char 93 at position 798
Please report this error to the GitHub page: https://github.com/JordanSantiagoYT/FNF-PsychEngine-NoBotplayLag

> Crash Handler written by: sqirra-rng
The engine has saved a crash log inside the crash folder, If you're making a GitHub issue you might want to send that!

thx for helping

**Comment by NAEL2XD** [comment, marked as answer]
Json not formatted correctly

**Reply by MryeetFN** [reply]
how do i fix that?
i started yesterday i dont rlly know what im doing

**Reply by NAEL2XD** [reply]
Go to the unformatted json and send it to me so I can fix it

**Reply by MryeetFN** [reply]
[masktricky.json](https://github.com/user-attachments/files/16458100/masktricky.json)

yes i am using tricky to learn

**Reply by NAEL2XD** [reply]
Replace the json with this:

```json
{
	"animations": [
		{
			"offsets": [
				-5,
				-1
			],
			"loop": false,
			"fps": 24,
			"anim": "idle",
			"indices": [],
			"name": "Idle0000"
		},
		{
			"offsets": [
				5,
				-6
			],
			"loop": false,
			"fps": 24,
			"anim": "singLEFT",
			"indices": [],
			"name": "Sing Left0000"
		},
		{
			"offsets": [
				-20,
				-51
			],
			"loop": false,
			"fps": 24,
			"anim": "singDOWN",
			"indices": [],
			"name": "SingDown0000"
		},
		{
			"offsets": [
				-46,
				27
			],
			"loop": false,
			"fps": 24,
			"anim": "singUP",
			"indices": [],
			"name": "Sing Up0000"
		},
		{
			"offsets": [
				-48,
				-7
			],
			"loop": false,
			"fps": 24,
			"anim": "singRIGHT",
			"indices": [],
			"name": "Sing Right0000"
		}
	],
	"vocals_file": "bf",
	"losing_colors": [
		25,
		124,
		0
	],
	"isPlayer": false	,
	"no_antialiasing": false,
	"image": "characters/trickymask",
	"shake_duration": 0,
	"position": [
		0,
		350
	],
	"health_drain": false,
	"shake_intensity": 0,
	"healthicon": "masktrickyicons",
	"shake_screen": false,
	"drain_floor": 0,
	"flip_x": true,
	"healthbar_colors": [
		25,
		124,
		0
	],
	"drain_amount": 0,
	"camera_position": [
		0,
		0
	],
	"sing_duration": 21,
	"noteskin": null,
	"winning_colors": [
		25,
		124,
		0
	],
	"scale": 1
}

**Reply by MryeetFN** [reply]
ok it works but down doesnt work and it doesnt animate?

**Reply by MryeetFN** [reply]
let me give the updated one with right offsets
[Uploading masktricky.json…]()

**Reply by NAEL2XD** [reply]
if down animation doesn't work that would mean you set the xml wrong, to fix it go find the down animation on xml and copy the name of the animation and place into the character animation for "singDOWN"

**Reply by MryeetFN** [reply]
i got the down animations working, but the thing is it doesnt animate it just shows the first frame of the picture

https://github.com/user-attachments/assets/ecfe84b5-6773-45b9-9711-db6437207c7b

**Reply by NAEL2XD** [reply]
You either:
- messed with the xml
Or 
- got a character with only 1 animation frame for each pose (idle, singLEFT, etc.)

**Reply by MryeetFN** [reply]
checked both over and over, dont see anything wrong, the xml is like the normal assets

**Reply by TheStinkern** [reply]
exclude the 0's on the xml animations, that should fix it
